src/api/middleware/controllers/book.js

```diff
@@ -0,0 +1,88 @@
+const Book = require("../../../models/Book");
+const { isValidObjectId } = require("../../../utils.js/validator");
+
+module.exports = {
+  create: async (req, res) => {
+    const {
+      title,
+      description,
+      subject,
+      total_quantity,
+      dimension,
+      total_price,
+      authorInformation,
+    } = req.body;
+
+    if (!isValidObjectId(authorInformation))
+      return res.json({ message: "authorInformation is invalid " });
+    if (!title) return res.json({ message: "title is required" });
+    if (!description) return res.json({ message: "description is required" });
+    if (!subject) return res.json({ message: "subject is required" });
+    if (!total_quantity)
+      return res.json({ message: "total_quantity is required" });
+    if (!dimension) return res.json({ message: "dimension is required" });
+    if (!total_price) return res.json({ message: "total_price is required" });
+
```

Comment:
How will you check if the total_price, total_quantity are invalid i.e. negative, not a number and/or empty spaces. You need learn how to use a validation framework(Joi) to avoid doing this manual validations, there are tedious.